commons-numbers-complex-arrays/src/main/java/org/apache/commons/numbers/complex/arrays/ComplexList.java

```diff
@@ -0,0 +1,357 @@
+
+import org.apache.commons.numbers.complex.Complex;
+
+import java.util.AbstractList;
+import java.util.Arrays;
+import java.util.Collection;
+
+/**
+ * Resizable-double array implementation of the List interface. Implements all optional list operations,
+ * and permits all elements. In addition to implementing the List interface,
+ * this class provides methods to manipulate the size of the array that is used internally to store the list.
+ *
+ * <p>Each ComplexList instance has a capacity. The capacity is half the size of the double array used to store the elements
+ * in the list. It is always at least twice as large as the list size. As elements are added to an ComplexList,
+ * its capacity grows automatically.</p>
+ *
+ * <p>An application can increase the capacity of an ComplexList instance before adding a large number of elements
+ * using the ensureCapacity operation. This may reduce the amount of incremental reallocation.</p>
+ */
+public class ComplexList extends AbstractList<Complex> {
+
+    /** The maximum size of array to allocate.
+     * Ensuring Max capacity is even with additional space for vm array headers.
+     */
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 9;
+
+    /** Max capacity for size of complex numbers in the list. */
+    public static final int MAX_CAPACITY = MAX_ARRAY_SIZE / 2;
+
+    /** error in case of allocation above max capacity. */
+    private static final String OOM_ERROR_STRING = "cannot allocate capacity %s greater than max " + MAX_CAPACITY;
+
+    /** Default initial capacity. */
+    private static final int DEFAULT_CAPACITY = 8;
+
+    /** Size label message. */
+    private static final String SIZE_MSG = ", Size: ";
+    /** Index position label message. */
+    private static final String INDEX_MSG = "Index: ";
+
+    /**
+     * The double array buffer into which the elements of the ComplexList are stored.
+     */
+    protected double[] realAndImagParts;
+
+    /**
+     * Size of ComplexList.
+     */
+    private int size;
+
+    /**
+     * Constructs an empty list with the specified capacity, if it's
+     * greater than the default capacity of 8.
+     *
+     * @param capacity - Capacity of list.
+     * @throws OutOfMemoryError - if the {@code capacity} is greater than {@code MAX_CAPACITY}.
+     */
+    public ComplexList(int capacity) {
+        if (capacity > MAX_CAPACITY) {
+            throw new OutOfMemoryError(String.format(OOM_ERROR_STRING, capacity));
+        }
+        final int arrayLength = Math.max(DEFAULT_CAPACITY, capacity) * 2;
+        realAndImagParts = new double[arrayLength];
+    }
+
+    /**
+     * Constructs an empty list with the default capacity of 8.
+     */
+    public ComplexList() {
+        realAndImagParts = new double[DEFAULT_CAPACITY * 2];
+    }
+
+    /**
+     * Gives the size of this ComplexList.
+     *
+     * @return the number of elements it contains.
+     */
+    @Override
+    public int size() {
```

Comment:
You should check if this javadoc provides anything useful over the default javadoc in List. If not then remove it. The same applies to all List methods in the interface. You can use the `{@inheritDoc}` tag to pull in the parent javadoc then add to it any further information.